### PR TITLE
Scss normalization

### LINKED ISSUE #2839: app settings dialog visibility should be increased
## Expected behavior

If an app provides a settings dialog, its visibility should be increased by using a different color or by separating the dialog with a small line etc.

Mookup 1:
20161223_nc_settings_1_new
Mookup 2:
20161223_nc_settings_2_new

## Current behavior

If an app provides a setting dialog, it is only indicated by a gear symbol and the word 'settings' which make difficult to see for many people. Additionally an opened dialog is not separated clear enough from underlying information which makes it also difficult to see where the dialog exactly begins.

Example 1:
20161223_nc_settings_1
Example 2:
20161223_nc_settings_2

### Steps to reproduce
1. Open the files, calender or contacts app
2. Check how the settings dialog is displayed.

#### Server Configuration

OS: Linux 3.2.82
Web server: Apache2 2.4.25
Database: MariaDB 5.5.53
PHP version: 5.6.23
Nextcloud version: 10.0.2

#### Client Configuration

Browser: Mozilla Firefox 50.1.0
Operating system: Windows 7

### PR DESCRIPTION
**Pull request by skjnldsv**
Processus:
1. Set the variables to
```
$color-main-text: #e2e2e2;
$color-main-background: #242424;
$color-primary: #0082c9;
$color-primary-text: #ffffff;
$color-error: #e9322d;
$color-warning: #ffcc44;
$color-success: #46ba61;
$color-primary-element: $color-primary;

@function nc-darken($color, $value) {
  @return lighten($color, $value);
}

@function nc-lighten($color, $value) {
  @return darken($color, $value);
}
```
2. Browse nextcloud and detect inconsistencies! :)

This pr includes:
 - Removal of the unused css filters
 - More variables integrated into server
 - Removal of undesired backgrounds
 - Increased visibility of the settings button (fix #2839)
 - Fixed the files busy state loader (from gif to css)
 - A lot of cleanups

### CODE REVIEW

File: core/css/apps.scss [hunk 1]
```diff
@@ -90,7 +90,7 @@ kbd {
 	-moz-user-select: none;
 	-ms-user-select: none;
 	user-select: none;
-	border-right: 1px solid nc-darken($color-main-background, 8%);
+	border-right: 1px solid $color-border;
```
Comment (MorrisJobke) [inline core/css/apps.scss]: Where does the `$color-border` comes from?

Comment (skjnldsv) [reply core/css/apps.scss]: Newly created var for all our borders in nc! :)

Comment (skjnldsv) [reply core/css/apps.scss]: It's a WIP, still haven't pushed the variables.scss 👍

**Comment by skjnldsv** [comment]
@nextcloud/designers what do you think of this?
kazam_screenshot_00002kazam_screenshot_00003

I always find the lack of feedback on the settings button disturbing, considering we have a unified design for all the links above :)

**Comment by jancborchardt** [comment]
@skjnldsv good call! Totally didn't occur to me :D

(I guess you are already thinking/working how to make the sidebar icons adapt to the background if in dark mode? ;) cc @juliushaertl

**Comment by skjnldsv** [comment]
@jancborchardt The switch to dark is the beginning of my whole work on the scss implementation, It never left my todo list!! :D 

Look at #1692 ;)

**Comment by skjnldsv** [comment]
Rebased to latest master
Will continue the work!

**Comment by skjnldsv** [comment]
I think we're good here! :)
Please review @nextcloud/designers

**Comment by MorrisJobke** [comment]
I will review once #6607 is fixed.

**Review by juliusknorr — APPROVED** [review]
Changes look good and i could not find any visual issues with it. :+1: 

By the way, it would be awesome if you could clean up the commit history for this a bit. Either squash it into one or even better group them somehow logically together.

**Comment by skjnldsv** [comment]
@juliushaertl why? :)

**Comment by juliushaertl** [comment]
@skjnldsv In general they are fine, since they are already nicely grouped by small changes. The main issue i have with those is, that most of the commit messages won't help you to get what this commit is about, when you read them without context. Makes it kind of hard to track down issues afterwords imo. Maybe just a personal thing.

**Comment by skjnldsv** [comment]
@juliushaertl I see your point! :)
I can squash all of it since this is a simple pr!

**Comment by skjnldsv** [comment]
@juliushaertl all clear! ;)

**Comment by juliushaertl** [comment]
Awesome, thanks. :+1: that also makes it easier to review and checkout the changes.

**Review by MorrisJobke — APPROVED** [review]

**Comment by MorrisJobke** [comment]
But somehow JSUnit fails:

```
PhantomJS 2.1.1 (Linux 0.0.0) OCA.Files.FileList tests Renaming files Restores thumbnail when rename was cancelled FAILED
	Expected 'http://localhost/core/img/filetypes/text.svg' to equal 'http://localhost/core/img/loading.gif'.
	apps/files/tests/js/filelistSpec.js:753:13
PhantomJS 2.1.1 (Linux 0.0.0) OCA.Files.FileList tests Moving files Restores thumbnail if a file could not be moved FAILED
	Expected 'http://localhost/core/img/filetypes/text.svg' to equal 'http://localhost/core/img/loading.gif'.
	apps/files/tests/js/filelistSpec.js:841:13
PhantomJS 2.1.1 (Linux 0.0.0) OCA.Files.FileList tests Copying files Restores thumbnail if a file could not be copied FAILED
	Expected 'http://localhost/core/img/filetypes/text.svg' to equal 'http://localhost/core/img/loading.gif'.
	apps/files/tests/js/filelistSpec.js:938:13
PhantomJS 2.1.1 (Linux 0.0.0) OCA.Files.FileList tests showFileBusyState shows spinner on busy rows FAILED
	Expected 'http://localhost/core/img/filetypes/image.svg' to equal 'http://localhost/core/img/loading.gif'.
	apps/files/tests/js/filelistSpec.js:3153:13
```

**Comment by skjnldsv** [comment]
Let me take a look @MorrisJobke

**Comment by skjnldsv** [comment]
@MorrisJobke Tests fixed. Let's wait for drone! 🎉

**Review by jancborchardt — APPROVED** [review]
:rocket:

**Pull request merged**